biomodels/src/main/java/org/reactome/release/Main.java

```diff
@@ -0,0 +1,212 @@
+package org.reactome.release;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.InstanceDisplayNameGenerator;
+import org.gk.model.PersistenceAdaptor;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+import org.gk.schema.InvalidAttributeValueException;
+import org.gk.schema.SchemaClass;
+import org.gk.util.GKApplicationUtilities;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author jweiser
+ *
+ */
+public class Main {
+
+    private static Map<String, GKInstance> biomodelsInstances = new HashMap<>();
+    private static final Logger logger = LogManager.getLogger();
+
+    /**
+     * @param args Command line arguments for the biomodels program
+     * @throws SQLException Thrown if unable to connect to the configured Reactome database
+     */
+    public static void main(String[] args) throws SQLException {
+        logger.info("Running BioModels insertion");
+
+        String pathToResources = args.length > 0 ? args[0] : "biomodels/src/main/resources/config.properties";
+        String pathToModels2Pathways = args.length > 1 ? args[1] : "biomodels/src/main/resources/models2pathways.tsv";
+
+        Properties props = new Properties();
+        try {
+            props.load(new FileInputStream(pathToResources));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        MySQLAdaptor dba = getDBA(props);
+
+        // Create new instanceEdit in database to track modified pathways
+        long personId = Long.parseLong(props.getProperty("personId"));
+        GKInstance instanceEdit = createInstanceEdit(dba, personId, "BioModels reference database creation");
+
+        Map<String, List<String>> pathwayStableIdToBiomodelsIds =
+            ModelsTSVParser.parse(pathToModels2Pathways);
+
+        for (GKInstance pathway: getPathwaysWithBiomodelsIds(dba, pathwayStableIdToBiomodelsIds.keySet())) {
+            logger.info("Adding BioModels ids to pathway " + pathway.getExtendedDisplayName());
+            List<String> biomodelsIds = pathwayStableIdToBiomodelsIds.get(getStableIdentifier(pathway));
+            List<GKInstance> biomodelsDatabaseIdentifiers =
+                createBiomodelsDatabaseIdentifiers(biomodelsIds, instanceEdit, dba);
+
+            try {
+                pathway.addAttributeValue(ReactomeJavaConstants.crossReference, biomodelsDatabaseIdentifiers);
+                pathway.addAttributeValue(ReactomeJavaConstants.modified, instanceEdit);
```

Comment:
Will this overwrite any existing modified instance edits?  I forget if it's necessary to load the existing attribute values first or not.

Reply:
You were correct, they needed to be loaded first. I think maybe we should conduct a thorough review of the new code to make sure this kind of thing isn't happening anywhere

Reply:
Could a Unit Test be added for this?  

You could extract the `try` block to a method that takes a pathway instance, the list of biomodelsDatabaseIdentifiers, and the instance edit as the three parameters.

Then test by sending a mock pathway instance that already has one modified instance edit which finishes with two modified instance edits afterwards?